frontend-maven-plugin/src/main/java/com/github/eirslett/maven/plugins/frontend/mojo/InstallNodeAndNpmMojo.java

```diff
@@ -72,34 +83,47 @@ protected boolean skipExecution() {
     }
 
     @Override
-    public void execute(FrontendPluginFactory factory) throws InstallationException {
+    public void execute(FrontendPluginFactory factory) throws Exception {
         ProxyConfig proxyConfig = MojoUtils.getProxyConfig(session, decrypter);
         String nodeDownloadRoot = getNodeDownloadRoot();
         String npmDownloadRoot = getNpmDownloadRoot();
         Server server = MojoUtils.decryptServer(serverId, session, decrypter);
+
+        String nodeVersion = NodeVersionDetector.getNodeVersion(workingDirectory, this.nodeVersion, this.nodeVersionFile);
+
+        if (isNull(nodeVersion)) {
+            throw new LifecycleExecutionException("Node version could not be detected from a file and was not set");
```

Comment:
I wonder if we could avoid the exception clause generalisations a bit - isn't what we report below also an "installation" exception/problem more than a "LifecycleExecutionException"?

Reply:
The message is all we care about from an end user perspective. From a dev perspective using `-X` will make it obvious the LoC where this originated.

The exceptions are the method of delivering messages to the users